src/main/java/dev/openfga/OpenFga.java

```diff
@@ -74,11 +73,10 @@ public boolean check(String objectType, String objectId, String relation, String
                 .user(String.format("%s:%s", userType, userId))
                 .relation(relation)
                 ._object(String.format("%s:%s", objectType, objectId));
-
         try {
             return Boolean.TRUE.equals(fgaClient.check(body).get().getAllowed());
-        } catch (InterruptedException | FgaInvalidParameterException | ExecutionException e) {
-            throw new OpenFgaCheckException("Error performing FGA check", e);
+        } catch (InterruptedException | FgaInvalidParameterException | ExecutionException cause) {
+            throw exceptionHandler.handle(cause, "Error performing FGA check");
```

Comment:
@jimmyjames Please take note of this line, this is the essential change in the PR.